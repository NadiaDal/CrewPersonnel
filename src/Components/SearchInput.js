// @flow

import React from 'react';

type SearchInputType = {
  placeholder: string,
  search: string,
  handleChange: (value: string) => void,
};

const SearchInput = ({ placeholder, search, handleChange }: SearchInputType) => (
  <input
    type="text"
    placeholder={placeholder}
    value={search}
    onChange={({ target: { value } }) => handleChange(value)}
    style={styles.input}
  />
);

const styles = {
  input: {
    width: '300px',
    maxWidth: '100%',
    backgroundColor: 'rgba(241, 241, 241, 0.8)',
    padding: '10px',
    margin: '10px',
    wordBreak: 'break-all',
    borderRadius: '7px',
  },
};

export default SearchInput;
