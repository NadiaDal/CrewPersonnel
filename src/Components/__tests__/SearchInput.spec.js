import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from '../SearchInput';

it('search input by name renders correctly', () => {
  const placeholder = 'Search by name';
  const search = 'Bra';
  const identity = () => {};

  const tree = renderer
    .create(<SearchInput placeholder={placeholder} search={search} handleChange={identity} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('search input by location renders correctly', () => {
  const placeholder = 'Search by city';
  const search = 'Nottingham';
  const identity = () => {};

  const tree = renderer
    .create(<SearchInput placeholder={placeholder} search={search} handleChange={identity} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
