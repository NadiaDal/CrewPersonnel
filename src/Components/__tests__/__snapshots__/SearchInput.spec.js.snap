// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input by location renders correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="Search by city"
  style={
    Object {
      "backgroundColor": "rgba(241, 241, 241, 0.8)",
      "borderRadius": "7px",
      "margin": "10px",
      "maxWidth": "100%",
      "padding": "10px",
      "width": "300px",
      "wordBreak": "break-all",
    }
  }
  type="text"
  value="Nottingham"
/>
`;

exports[`search input by name renders correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="Search by name"
  style={
    Object {
      "backgroundColor": "rgba(241, 241, 241, 0.8)",
      "borderRadius": "7px",
      "margin": "10px",
      "maxWidth": "100%",
      "padding": "10px",
      "width": "300px",
      "wordBreak": "break-all",
    }
  }
  type="text"
  value="Bra"
/>
`;
